src/sagemaker/estimator.py

```diff
@@ -1478,12 +1478,13 @@ def __init__(
                     >>>         |----- test.py
 
                     You can assign entry_point='src/train.py'.
-            source_dir (str): Path (absolute, relative, or an S3 URI) to a directory with
-                any other training source code dependencies aside from the entry
-                point file (default: None). Structure within this directory are
-                preserved when training on Amazon SageMaker. If 'git_config' is
-                provided, 'source_dir' should be a relative location to a
-                directory in the Git repo. .. admonition:: Example
+            source_dir (str): Path (absolute, relative, or an S3 URI points to a
+                tar.gz) to a directory with any other training source code
+                dependencies aside from the entry point file (default: None).
+                Structure within this directory are preserved when training on
+                Amazon SageMaker. If 'git_config' is provided, 'source_dir'
+                should be a relative location to a directory in the Git repo.
+                .. admonition:: Example
```

Comment:
I worry the way you've worded it might be a bit ambiguous, i.e. one might think even the absolute/relative paths should point to a `tar.gz` file. How about adding another sentence after "Structure within...Amazon SageMaker." that's like, "If `source_dir` is an S3 URI, it must point to a `tar.gz` file"?

Reply:
This is better!